### PR TITLE
Missing first last

### PR DESCRIPTION
**Pull request by tpendragon**

### CODE REVIEW

File: lib/active_fedora/aggregation/list_source.rb [hunk 1]
```diff
@@ -9,7 +9,14 @@ class ListSource < ActiveFedora::Base
       def save(*args)
         return true if has_unpersisted_proxy_for? || !changed?
         persist_ordered_self if ordered_self.changed?
-        super
+        super.tap do |result|
+          if result && args.first.kind_of?(Hash) && args.first[:container].present?
+            owner = args.first[:container]
+            owner.head = [head_id.first]
+            owner.tail = [tail_id.first]
+            owner.save
```
Comment (jcoyne) [inline lib/active_fedora/aggregation/list_source.rb]: Should this be save! Should the status be checked?

File: spec/activefedora/ordered_spec.rb [hunk 2]
```diff
@@ -71,10 +71,13 @@ class Image < ActiveFedora::Base
     describe "<<" do
       it "appends" do
         member = Member.new
+        subject.save
```
Comment (jcoyne) [inline spec/activefedora/ordered_spec.rb]: Can you use save! or check the result

File: spec/activefedora/ordered_spec.rb [hunk 3]
```diff
@@ -71,10 +71,13 @@ class Image < ActiveFedora::Base
...
         expect(subject.ordered_members << member).to eq [member]
         expect(subject.ordered_members).to eq [member]
         expect(subject.members).to eq [member]
         expect(subject.ordered_member_proxies.to_a.length).to eq 1
+        subject.save
```
Comment (jcoyne) [inline spec/activefedora/ordered_spec.rb]: Same here

File: lib/active_fedora/aggregation/list_source.rb (hunk 1 above)
Comment (tpendragon) [inline lib/active_fedora/aggregation/list_source.rb]: save! is probably appropriate.

**Comment by tpendragon** [comment]
Plum's test say with this branch it's not moving everything to members...not sure what the deal there is.

**Comment by tpendragon** [comment]
Okay, so calling save! before after_save stuff runs turned out to break auto-saving associations. This ensures that the members one runs at least, but could be breaking other auto-saves. Not terribly friendly, but I'm not sure what to do.

File: lib/active_fedora/orders/builder.rb [hunk 4]
```diff
@@ -22,10 +22,35 @@ def initialize(model, name, options)
...
+        end
+      end
+    end
+
     def build
-      super.tap do
+      super.tap do |result|
         model.property :head, predicate: ::RDF::Vocab::IANA['first']
         model.property :tail, predicate: ::RDF::Vocab::IANA.last
+        model.send(:define_method, :apply_first_and_last) do
```
Comment (jcoyne) [inline lib/active_fedora/orders/builder.rb]: Is there a reason that `apply_first_and_last` couldn't go into `FixFirstLast`

Comment (tpendragon) [inline lib/active_fedora/orders/builder.rb]: Because it needs metadata the builder has.

Comment (tpendragon) [inline lib/active_fedora/orders/builder.rb]: Specifically the reflection's :through option.

**Comment by jcoyne** [comment]
Please squash your commits and then it looks good.

**Comment by tpendragon** [comment]
@jcoyne Squashed

**Pull request merged**